### ISSUE TITLE
v0.3.4 fails to build with latest nightly

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rustfmt#2362]
````
rustc 1.25.0-nightly (3f92e8d89 2018-01-14)
cargo 0.25.0-nightly (a88fbace4 2017-12-29)
````

````
    Updating registry `https://github.com/rust-lang/crates.io-index`
  Installing rustfmt-nightly v0.3.4
   Compiling getopts v0.2.15
   Compiling regex-syntax v0.4.2
   Compiling rustc-demangle v0.1.5
   Compiling cfg-if v0.1.2
   Compiling term v0.4.6
   Compiling serde v1.0.27
   Compiling semver-parser v0.7.0
   Compiling itoa v0.3.4
   Compiling utf8-ranges v1.0.0
   Compiling num-traits v0.1.41
   Compiling cc v1.0.4
   Compiling dtoa v0.4.2
   Compiling quote v0.3.15
   Compiling unicode-segmentation v1.2.0
   Compiling unicode-xid v0.0.4
   Compiling diff v0.1.11
   Compiling void v1.0.2
   Compiling rustfmt-nightly v0.3.4
   Compiling lazy_static v1.0.0
   Compiling libc v0.2.36
   Compiling log v0.4.1
   Compiling semver v0.8.0
   Compiling toml v0.4.5
   Compiling serde_json v1.0.9
   Compiling unreachable v1.0.0
   Compiling synom v0.11.3
   Compiling backtrace-sys v0.1.16
   Compiling memchr v2.0.1
   Compiling log v0.3.9
   Compiling thread_local v0.3.5
   Compiling syn v0.11.11
   Compiling aho-corasick v0.6.4
   Compiling regex v0.2.5
   Compiling backtrace v0.3.5
   Compiling error-chain v0.11.0
   Compiling env_logger v0.4.3
   Compiling serde_derive_internals v0.19.0
   Compiling derive-new v0.5.0
   Compiling serde_derive v1.0.27
   Compiling cargo_metadata v0.4.1
error[E0599]: no variant named `AutoImpl` found for type `syntax::ast::ItemKind` in the current scope
   --> src/visitor.rs:413:13
    |
413 |             ast::ItemKind::AutoImpl(..) => {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `syntax::ast::ItemKind`

error: aborting due to previous error

error: failed to compile `rustfmt-nightly v0.3.4`, intermediate artifacts can be found at `/tmp/cargo-install.ErWlQh8PEJgz`

Caused by:
  Could not compile `rustfmt-nightly`.

To learn more, run the command again with --verbose.
````

Building from the repo  @ 74a957e80f4808227d41903876a9e2ef66438910 works however.

**Comment by topecongiro** [comment]
Recently we switched to `rustc-ap-syntax` crate over libsyntax tied with rustc, so the latest master is not affected by the breaking change to libsyntax. 0.3.4 still uses libsyntax tied with rustc, hence the build failure.

**Comment by topecongiro** [comment]
cc https://github.com/rust-lang/rust/pull/47416.

**Comment by topecongiro** [comment]
Unfortunately, the latest `rustc-ap-syntax` (11.0.0) does not include the relevant commits, so we must wait until 12.0.0 gets published. It will be published within 24 hours.

**Comment by topecongiro** [comment]
Published 0.3.5.
